payu/models/cice.py

```diff
@@ -333,6 +342,50 @@ def archive(self, **kwargs):
         else:
             shutil.rmtree(self.work_input_path)
 
+        if self.compression_enabled():
+            self.compress_log_files()
+
+    def compression_enabled(self):
+        """
+        Determine whether to run log compression based on config.yaml settings.
+        Default to True when 'compress_logs' setting is absent.
+        """
+        archive_config = self.expt.config.get('archive', {})
+        if isinstance(archive_config, dict):
+            return archive_config.get('compress_logs', True)
+        else:
+            return True
+
+    def get_log_files(self):
+        """
+        Find model log files in the work directory based on regex patterns
+        in self.logs_to_compress.
+
+        Returns
+        -------
+        log_files: list of paths to model log files.
+        """
+        log_files = []
+        for filename in os.listdir(self.work_path):
+            if any((re.match(pattern, filename)
+                    for pattern in self.logs_to_compress)):
```

Comment:
An alternative would be to create a single regex with all your patterns separate by `OR` (`|`) and matching once for each file, e.g. https://stackoverflow.com/a/8888615

Reply:
Good idea, I've added this in d110cae